pkg/csi/cinder/controllerserver.go

```diff
@@ -394,6 +476,61 @@ func (cs *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateS
 		klog.Errorf("Failed to WaitSnapshotReady: %v", err)
 		return nil, status.Error(codes.Internal, fmt.Sprintf("CreateSnapshot failed with error %v", err))
 	}
+	// If snapshotType is 'backup', create a backup from the snapshot and delete the snapshot.
+	if snapshotType == "backup" {
+		// Add cluster ID to the snapshot metadata
+		properties := map[string]string{cinderCSIClusterIDKey: cs.Driver.cluster}
+
+		// see https://github.com/kubernetes-csi/external-snapshotter/pull/375/
+		// Also, we don't want to tag every param but we still want to send the
+		// 'force-create' flag to openstack layer so that we will honor the
+		// force create functions
+		for _, mKey := range []string{"csi.storage.k8s.io/volumesnapshot/name", "csi.storage.k8s.io/volumesnapshot/namespace", "csi.storage.k8s.io/volumesnapshotcontent/name", openstack.SnapshotForceCreate, openstack.SnapshotType} {
+			if v, ok := req.Parameters[mKey]; ok {
+				properties[mKey] = v
+			}
+		}
+		backup, err = cs.Cloud.CreateBackup(name, volumeID, snap.ID, &properties)
```

Comment:
My mistake. In case of a snapshot it works because here it will detect the volumeID being the same
```
	if len(snapshots) == 1 {
		snap = &snapshots[0]

		if snap.VolumeID != volumeID {
			return nil, status.Error(codes.AlreadyExists, "Snapshot with given name already exists, with different source volume ID")
		}
		klog.V(3).Infof("Found existing snapshot %s from volume with ID: %s", name, volumeID)
}
```

i will change the logic so that it also does this check before trying to create the backup again.
It will go directly to `waitBackupReady()` if a backup with the same volumeID and name exists.

Reply:
There should be no issues with cinder-csi restarts now